### PR TITLE
add idp negative tests - validate mandatory fields

### PR DESCRIPTION
**Pull request by shellymiron**
This PR will validate all the mandatory fields each idp resource has, as described in the[ terraform rhcs provider doc](https://registry.terraform.io/providers/terraform-redhat/rhcs/latest/docs/resources/identity_provider)

image

### CODE REVIEW

**Comment by cmyster** [comment]
LGTM

File: tests/ci/labels.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 // Features
 var FeatureMachinepool = Label("feature-machinepool")
 var FeatureIDP = Label("feature-idp")
+var FeatureIDPNegative = Label("feature-idp-negative")
```
Comment (amalykhi) [inline tests/ci/labels.go:10]: I don't think we need additional label for this

File: tests/e2e/idps_test.go [hunk 2]
```diff
@@ -29,240 +29,473 @@ var _ = Describe("TF Test", func() {
...
+				})
 			})
+			Context("LDAP IDP test cases", func() {
+				BeforeEach(func() {
 
-			AfterEach(func() {
-				err := idpService.htpasswd.Destroy()
-				Expect(err).ToNot(HaveOccurred())
+					userName = "newton"
+					password = "password"
```
Comment (amalykhi) [inline tests/e2e/idps_test.go:102]: I wouldn't use a hardcoded password even if it's a negative test case.

Comment (shellymiron) [reply tests/e2e/idps_test.go:102]: this specific password must be hardcoded, this is the password for ldap test cases when we need to use 
`"ldap://ldap.forumsys.com/dc=example,dc=com?uid"`

File: tests/ci/labels.go (hunk 1 above)
Comment (shellymiron) [reply tests/ci/labels.go:10]: ok, so maybe later on we will add a general "negative-tc" label? so we could run all the negative tests at once?
or should each feature run all it's cases without negative/positive consideration?

File: tests/e2e/idps_test.go (hunk 2 above)
Comment (cmyster) [reply tests/e2e/idps_test.go:102]: (just a suggestion) On the other hand, we can set it as a setting and then use the same code for positive ("newton") and negative test ("notnewton") or something trickier such as a username injection attack.

Comment (shellymiron) [reply tests/e2e/idps_test.go:102]: sure, I plan to do several negative tests in the future
I'll insert those aswell

File: tests/utils/exec/idps.go [hunk 3]
```diff
@@ -51,7 +52,7 @@ func (idp *IDPService) Init(manifestDirs ...string) error {
 }
 
 func (idp *IDPService) Create(createArgs *IDPArgs, extraArgs ...string) error {
-	createArgs.URL = CON.GateWayURL
+	// createArgs.URL = CON.GateWayURL
```
Comment (amalykhi) [inline tests/utils/exec/idps.go:55]: Why commented?

Comment (shellymiron) [reply tests/utils/exec/idps.go:55]: `createArgs.URL` is not necessary in order to run the args as Iv'e seen.
but moreover - when we do use it, and there is an idp which require its own URL, the test fail as duplication.

Comment (amalykhi) [reply tests/utils/exec/idps.go:55]: Let's delete it is it not necessary and duplication?

Comment (shellymiron) [reply tests/utils/exec/idps.go:55]: ack, deleting

**Pull request merged**